packages/braintree-utils/src/braintree.ts

```diff
@@ -883,6 +884,212 @@ export interface BraintreeConnectCardComponent {
+    BraintreeFastlaneConfig
+>;
+
+export interface BraintreeFastlaneConfig {
+    authorization: string;
+    client: BraintreeClient;
+    deviceData?: string;
+    styles?: BraintreeFastlaneStylesOption;
+}
+
+export interface BraintreeFastlane {
+    identity: BraintreeFastlaneIdentity;
+    FastlaneCardComponent: (
+        options: BraintreeFastlaneCardComponentOptions,
+    ) => BraintreeFastlaneCardComponent;
+    events: BraintreeFastlaneEvents;
+}
+
+export interface BraintreeFastlaneWindow extends Window {
+    braintreeFastlane: BraintreeFastlane;
+}
+
+export interface BraintreeFastlaneIdentity {
+    lookupCustomerByEmail(email: string): Promise<BraintreeFastlaneLookupCustomerByEmailResult>;
+    triggerAuthenticationFlow(
+        customerId: string,
+        options?: BraintreeFastlaneAuthenticationOptions,
+    ): Promise<BraintreeFastlaneAuthenticationCustomerResult>;
+}
+
+export interface BraintreeFastlaneLookupCustomerByEmailResult {
+    customerContextId?: string;
+}
+
+export interface BraintreeFastlaneAuthenticationOptions {
+    styles?: BraintreeFastlaneStylesOption;
+    fetchFullProfileData?: boolean; // default: true
+}
+
+export interface BraintreeFastlaneStylesOption {
+    root?: {
+        backgroundColorPrimary?: string;
+        errorColor?: string;
+        fontFamily?: string;
+    };
+    input?: {
+        borderRadius?: string;
+        borderColor?: string;
+        focusBorderColor?: string;
+    };
+    toggle?: {
+        colorPrimary?: string;
+        colorSecondary?: string;
+    };
+    text?: {
+        body?: {
+            color?: string;
+            fontSize?: string;
+        };
+        caption?: {
+            color?: string;
+            fontSize?: string;
+        };
+    };
+    branding?: string; // 'light' | 'dark'
+}
+
+export enum BraintreeFastlaneAuthenticationState {
+    SUCCEEDED = 'succeeded',
+    FAILED = 'failed',
+    CANCELED = 'cancelled',
+    UNRECOGNIZED = 'unrecognized',
+}
+
+export interface BraintreeFastlaneAuthenticationCustomerResult {
+    authenticationState: BraintreeFastlaneAuthenticationState;
+    profileData: BraintreeFastlaneProfileData;
+}
+
+export interface BraintreeFastlaneProfileData {
+    fastlaneCustomerAuthAssertionToken: string;
+    fastlaneCustomerId: string;
+    shippingAddress: BraintreeFastlaneAddress;
+    card: BraintreeFastlaneVaultedInstrument;
+    name: {
+        given_name: string;
+        surname: string;
+    };
+}
+
+export interface BraintreeFastlaneAddress {
+    id?: string;
+    firstName?: string;
+    lastName?: string;
+    company?: string;
+    streetAddress: string;
+    extendedAddress?: string;
+    locality: string;
+    region: string;
+    postalCode: string;
+    countryCodeNumeric?: number;
+    countryCodeAlpha2: string;
+    countryCodeAlpha3?: string;
+}
+
+export interface BraintreeFastlaneCardPaymentSource {
+    brand: string;
+    expiry: string;
+    lastDigits: string;
+    name?: string;
```

Comment:
Are you sure that we will have name inside? Should name be a string?

Reply:
Yes, in their docs name is string inside CardPaymentSource